cpp/src/arrow/ipc/metadata-internal.cc

```diff
@@ -246,6 +246,16 @@ Status FieldFromFlatbuffer(const flatbuf::Field* field, std::shared_ptr<Field>*
 
 // Implement MessageBuilder
 
+// will return the endianness of the system we are running on
+flatbuf::Endianness endianness() {
+  union {
+    uint32_t i;
+    char c[4];
+  } bint = {0x01020304};
+
+  return bint.c[0] == 1 ? flatbuf::Endianness_Big : flatbuf::Endianness_Little;
```

Comment:
Here is a suitable reference for this code https://github.com/numpy/numpy/blob/master/numpy/core/src/multiarray/multiarraymodule.c#L2910 — is there another source where it might have come from?